group-coordinator/src/test/java/org/apache/kafka/coordinator/group/GroupMetadataManagerTest.java

```diff
@@ -376,19 +448,28 @@ public CoordinatorResult<ConsumerGroupHeartbeatResponseData, Record> consumerGro
             return result;
         }
 
-        public List<MockCoordinatorTimer.ExpiredTimeout<Record>> sleep(long ms) {
+        public List<ExpiredTimeout<Void, Record>> sleep(long ms) {
             time.sleep(ms);
-            List<MockCoordinatorTimer.ExpiredTimeout<Record>> timeouts = timer.poll();
-            timeouts.forEach(timeout -> timeout.records.forEach(this::replay));
+            List<ExpiredTimeout<Void, Record>> timeouts = timer.poll();
+            timeouts.forEach(timeout -> {
+                if (timeout.result.replayRecords()) {
+                    timeout.result.records().forEach(this::replay);
+                }
+            });
             return timeouts;
         }
 
-        public MockCoordinatorTimer.ScheduledTimeout<Record> assertSessionTimeout(
+        public void sleepAndAssertEmptyResult(long ms) {
+            List<ExpiredTimeout<Void, Record>> timeouts = sleep(ms);
+            timeouts.forEach(timeout -> assertEquals(EMPTY_RESULT, timeout.result));
+        }
```

Comment:
It seems based on the usages of this method that only one timeouts is expected all the time. Should we enforce it as well?

More generally, I was wondering if having a `assertEmptyTimeout` helper method and using `assertEmptyTimeout(context.sleep(...))` would have a better separation of concerns. I leave this up to you.

Reply:
will keep it as assertEmptyResult as the timeout is not empty (can be) but we want to assert that the coordinator result is.